pkg/runtime.v2/framework/interface.go

```diff
@@ -53,5 +53,5 @@ type CustomValidationPlugin interface {
 
 type ComponentBuilderPlugin interface {
 	Plugin
-	Build(ctx context.Context, info *runtime.Info, trainJob *kubeflowv2.TrainJob) (client.Object, error)
+	Build(ctx context.Context, runtimeJobTemplateSpec interface{}, info *runtime.Info, trainJob *kubeflowv2.TrainJob) (client.Object, error)
```

Comment:
```suggestion
	Build(ctx context.Context, obj client.Object, info *runtime.Info, trainJob *kubeflowv2.TrainJob) (client.Object, error)
```

As I mentioned in another thread, we should provide the guardrail.

Reply:
Resolved.